README.md

```diff
@@ -662,6 +664,23 @@ type SomeObject = NumberDictionary<{ a: number; b: string }>;
 type EmptyObject = NumberDictionary<{}>;
 ```
 
+### NonEmptyArray
+
+Useful for accepting only arrays containing at least one element.
+
+```typescript
+type NonEmptyRestParameters = NonEmptyArray<number>;
+// Result: [number, ...number[]]
+
+// declare function expression type accepting some rest parameters, but at least one element for the rest parameters is required
+type FunctionAcceptingRestParameters = (someString: string, ...args: NonEmptyRestParameters<number>) => void;
```

Comment:
Would be cool to have this feature in TypeScript, but it's not implemented yet 😅

```suggestion
type FunctionAcceptingRestParameters = (someString: string, ...args: NonEmptyRestParameters) => void;
```

Reply:
Ah damn, I think my original idea was to make NonEmptyRestParameters generic :D